ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift

```diff
@@ -357,9 +357,17 @@ extension KeymanWebViewController {
   }
   
   func setBannerHeight(to height: Int) {
-    // TODO:
     webView?.evaluateJavaScript("setBannerHeight(\(height));", completionHandler: nil)
   }
+
+  /**
+   * Ensures that the embedded KMW instance uses the app's current error-report toggle setting.
+   * This is always called during keyboard page initialization and may also be called any time
+   * thereafter for settings updates.
+   */
+  func setSentryState(enabled: Bool = SentryManager.enabled) {
+    webView?.evaluateJavaScript("sentryManager.enabled = \(enabled ? "true" : "false")")
```

Comment:
Is there a chance this function can be called when the webView is not ready to evaluate the JS (because page not fully loaded or other reasons)?

Reply:
Yes.  This is _exactly_ why _this_ part exists:

https://github.com/keymanapp/keyman/blob/193813903898fc1fccd1e336d9f99252ab9b0990/ios/engine/KMEI/KeymanEngine/Classes/KeymanWebViewController.swift#L570-L577

Emphasis on the final line.

Reply:
That ensures that it's called as soon as the page _is_ ready, in case the function had been called when it was _not_ ready.

Reply:
So if the function is called before the page is ready (via Swift's `SentryManager`), we'll get a spurious js error in the console log? I guess that's okay, just a shame. Could wrap it in a js try/catch ;-)

Reply:
Logged as #4125